sql/hive/src/main/java/org/apache/hadoop/hive/ql/io/orc/SparkOrcNewRecordReader.java

```diff
@@ -42,8 +41,11 @@
 
   public SparkOrcNewRecordReader(Reader file, Configuration conf,
       long offset, long length) throws IOException {
-    List<OrcProto.Type> types = file.getTypes();
-    numColumns = (types.size() == 0) ? 0 : types.get(0).getSubtypesCount();
+    if (file.getTypes().size() == 0) {
```

Comment:
[HIVE-11890](https://issues.apache.org/jira/browse/HIVE-11890) move `org.apache.hadoop.hive.ql.io.orc.OrcProto.Type` to `org.apache.orc.OrcProto.Type`.
Remove `OrcProto.Type` to workaround this api change.

Reply:
Could even use `isEmpty` here

Reply:
Yes. `isEmpty` is better.